app/src/main/java/com/poupa/vinylmusicplayer/ui/fragments/player/card/CardPlayerFragment.java

```diff
@@ -450,7 +450,11 @@ public AnimatorSet createDefaultColorChangeAnimatorSet(int newColor) {
                 animatorSet.play(subHeaderAnimator);
             }
 
-            animatorSet.setDuration(ViewUtil.VINYL_MUSIC_PLAYER_ANIM_TIME);
```

Comment:
Instead of working around the AnimatorSet implementation and try to propagate the duration, would it make sense to move the line `animatorSet.setDuration` to line 443, i.e. just before the first `play` request on the set?

Reply:
Well what about concurrency - what if the user tries to do something at a faster rate than the animation duration. It gets complicated. We'd need a util method to check if animations are playing, stop them if so, set the animation duration, and then play the animation. If that's what you want I'm good with it, but that's what's required as far as I can tell

Reply:
Then again, if each animation uses a new animation set I believe the default behavior would just be to layer the effects of them, which would not crash the app, but might result in visual bugs, so again best to stop then duration then play if that's what you want

Reply:
> Well what about concurrency - what if the user tries to do something at a faster rate than the animation duration. It gets complicated. We'd need a util method to check if animations are playing, stop them if so, set the animation duration, and then play the animation. If that's what you want I'm good with it, but that's what's required as far as I can tell

Not sure that I understand your point, and what the bug is. So please correct me if I'm wrong.

This class creates a new AnimSet object, attach multiple animations to it (via `play`, `with`, etc) and then `start` them all. In between, we `setDuration` to give the time bounds to the set.

1. Since the AnimSet object is freshly created and not started, I dont get why we are frowned upon by Android telling that anim is already started.
2. Assuming that your approach is correct, why this is not the default behaviour of AnimSet.setDuration? Or if the behaviour depends on the OS version, we should have a AnimCompat utility class to take care of this for us.
3. If there is ordering constraint between the animation, ie play(A).before(B) or the complete set of animation is not defined yet, then the fact that we inlined the `setDuration` with a loop is not entirely correct, right? Probably this is what you suggested with "check if animations are playing, stop them if so, set the animation duration, and then ~play~ start the animation"?
4. Since we only experience this crash on Android 12+, would it make sense to wrap it with `if (API >= 31)`?

I've applied your changes to my fork https://github.com/vinyl2-team/vinyl2. So far so good since few hours.

Reply:
1. I'm not Android, obviously this is a common pattern and everyone is confused. According to the bug it started being a think at Android 8.
2. My fix was to have functionally equivalent code to the original that doesn't crash the app. This isn't the default because the android team prefers to crash in this instance.
3. Yeah sounds like you're for that solution, probably more robust than "just mimic existing code but don't crash". I can look into this but it will be slower to create than this quick fix.
4. That would disable animations during 13+ by making them zero length, so not ideal. I'd rather not special case code unless strictly necessary, and this case it is but necessary.

Reply:
Also I really don't have a clear picture on why this code crashed originally, I just wanted to write a functional equivalent code that didn't crash

Reply:
> I can look into this but it will be slower to create than this quick fix.

Please do, since you can reproduce easily. Thanks!

> That would disable animations during 13+ by making them zero length, so not ideal

We can do away with if/else, no? 😀

Reply:
On second though, I'm OK to stick with this implementation, which is simple. Unless you have a better option, the changes we discussed would result in quite complicated code.

Please tell me.

Reply:
Yes the code would be complex, I am willing to go with this simple code as well just so the app doesn't crash